### PR TITLE
docs(MADR): mesh service api

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Checklist prior to review

MADR for MeshService API.

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by lahabana — APPROVED** [review]

**Review by michaelbeaumont — APPROVED** [review]

File: docs/madr/decisions/039-meshservice-api.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+kind: MeshService
+metadata:
+  name: redis
+  namespace: redis-system
+  labels: # you could then select this in when you use `MeshService` in `to` section
+    team: db-operators
+    kuma.io/mesh: default
+spec:
+  selector:
+    dataplaneTags: # tags in Dataplane object, see below
```
Comment (lobkovilya) [inline docs/madr/decisions/039-meshservice-api.md:59]: Is it necessary to nest `dataplaneTags` into `selector`? Can we use a k8s-like API:
```yaml
spec:
  selector:
    app: redis
    k8s.kuma.io/namespace: redis-system
    kuma.io/zone: east-1
```

File: docs/madr/decisions/039-meshservice-api.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  ports:
+  - port: 6739
+    targetPort:
+      value: 6739
+    appProtocol: tcp
+  - name: some-port
+    port: 16739
+    targetPort:
+      name: target-port-from-container # name of the inbound
+    appProtocol: tcp
```
Comment (lobkovilya) [inline docs/madr/decisions/039-meshservice-api.md:72]: k8s has both `appProtocol` and `protocol`, are we fine having only `appProtocol`?

File: docs/madr/decisions/039-meshservice-api.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+status: # managed by CP. Not shared cross zone, but synced to global
+  availability: Online # | Offline | NotAvailable | PartialyDegraded
+  tls:
+    state: Ready
+    issuedBackends:
+      ca-1: 5
+  proxies:
+    offline: 3
+    online: 5
+    total: 8
```
Comment (lobkovilya) [inline docs/madr/decisions/039-meshservice-api.md:82]: nit: do we need all 3? I guess when CP sets the status it's going to ensure anyway that `offline+online=total` right?

File: docs/madr/decisions/039-meshservice-api.md (hunk 2 above)
Comment (jakubdyszkiewicz) [reply docs/madr/decisions/039-meshservice-api.md:72]: yes, we care only about appProtocol

File: docs/madr/decisions/039-meshservice-api.md (hunk 1 above)
Comment (jakubdyszkiewicz) [reply docs/madr/decisions/039-meshservice-api.md:59]: I'd rather be explicit about this. I expect that in the future we will refactor Dataplane object to MeshProxy object that will have a one set of tags for all inbounds.

File: docs/madr/decisions/039-meshservice-api.md (hunk 3 above)
Comment (jakubdyszkiewicz) [reply docs/madr/decisions/039-meshservice-api.md:82]: I guess we don't

File: docs/madr/decisions/039-meshservice-api.md (hunk 1 above)
Comment (lobkovilya) [reply docs/madr/decisions/039-meshservice-api.md:59]: But this `selector` is a result of the conversion of the k8s service to MeshService? IIUC users can't change MeshService resources directly. So you'll need a way to create different selectors on k8s service, right?

Also if you replace some Dataplane objects with MeshProxy objects, don't you want `selector` to still work without changing `dataplaneTags` to `meshProxyTags`?

**Review by lobkovilya — APPROVED** [review]

**Pull request merged**